### PR TITLE
Refactor concourse chart

### PR DESCRIPTION
**Pull request by iuriaranda**
- Only install Concourse web/atc
- Use skyscrapers' docker image to pull keys from S3
- Refactor values structure to put everything under 'concourse'
- Added ingress for atc and tsa
- ~Added rbac settings~

TODO
- ~I still need to test everything~
- ~Figure out how to access the tsa port (SSH) without the nginx ingress (for worker registration)~
- ~Figure out rbac permissions~

### CODE REVIEW

File: concourse/templates/ingress.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    component: web
+    release: "{{ .Release.Name }}"
+    heritage: "{{ .Release.Service }}"
+    dns: route53
+  annotations:
+    kubernetes.io/ingress.class: "nginx"
+    kubernetes.io/tls-acme: "true"
+spec:
+  tls:
+  - secretName: {{ template "web.fullname" . }}-lego-tls
```
Comment (SamClinckspoor) [inline concourse/templates/ingress.yaml]: I think this secret is set in the ingress automatically by kube-lego

File: concourse/templates/role.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+apiVersion: rbac.authorization.k8s.io/v1beta1
+kind: ClusterRole
+metadata:
+  name: {{ template "concourse.fullname" . }}
+  labels:
+    app: {{ template "name" . }}
+    chart: {{ .Chart.Name }}-{{ .Chart.Version }}
+    heritage: {{ .Release.Service }}
+    release: {{ .Release.Name }}
+rules: # TODO
```
Comment (SamClinckspoor) [inline concourse/templates/role.yaml]: does it even need a role? afaik it doesn't access anything cluster related

File: concourse/templates/web-deployment.yaml [hunk 3]
```diff
@@ -5,22 +5,29 @@ metadata:
   labels:
     app: {{ template "web.fullname" . }}
     chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    component: web
     release: "{{ .Release.Name }}"
     heritage: "{{ .Release.Service }}"
 spec:
```
Comment (SamClinckspoor) [inline concourse/templates/web-deployment.yaml:11]: You will need to specify the serviceaccount you created if you want to use it. (I don't think concourse needs serviceaccount/rbac)

File: concourse/templates/web-svc.yaml [hunk 4]
```diff
@@ -5,25 +5,26 @@ metadata:
...
     heritage: "{{ .Release.Service }}"
     dns: route53
   annotations:
-    domainName: {{ .Values.web.service.domainName }}
-    service.beta.kubernetes.io/aws-load-balancer-backend-protocol: http
-    service.beta.kubernetes.io/aws-load-balancer-cross-zone-load-balancing-enabled: "true"
-    service.beta.kubernetes.io/aws-load-balancer-ssl-cert: {{ .Values.web.service.acm-arn }}
+    kubernetes.io/ingress.class: "nginx"
+    kubernetes.io/tls-acme: "true"
+    external-dns.alpha.kubernetes.io/hostname: {{ .Values.concourse.web.service.domainName }}.
```
Comment (SamClinckspoor) [inline concourse/templates/web-svc.yaml]: these annotations are not needed, its handled by the ingress

File: concourse/templates/web-svc.yaml [hunk 5]
```diff
@@ -5,25 +5,26 @@ metadata:
...
 spec:
-  type: LoadBalancer
+  tls:
+  - secretName: {{ template "concourse.fullname" . }}-lego-tls
+    hosts:
+    - {{.Values.concourse.web.service.domainName}}
```
Comment (SamClinckspoor) [inline concourse/templates/web-svc.yaml]: same here

**Review by SamClinckspoor — COMMENTED** [review]
For SSH I think you can create a tcp ingress for 22

File: concourse/templates/ingress.yaml (hunk 1 above)
Comment (iuriaranda) [reply concourse/templates/ingress.yaml]: kube-lego creates the secret, but this is to let nginx know where to find the secret containing the certificate. At least this is how we set it up for kubesignin.

File: concourse/templates/role.yaml (hunk 2 above)
Comment (iuriaranda) [reply concourse/templates/role.yaml]: true, should I also remove the serviceaccount, or I leave the serviceaccount without any role associated?

File: concourse/templates/web-svc.yaml (hunk 4 above)
Comment (iuriaranda) [reply concourse/templates/web-svc.yaml]: true, this is a leftover, I'll remove it

**Comment by iuriaranda** [comment]
it seems that ingresses don't support tcp yet: https://kubernetes.io/docs/concepts/services-networking/ingress/#the-ingress-resource

Are there other ways we can route traffic to a service through tcp?

cc @SamClinckspoor

File: concourse/templates/role.yaml (hunk 2 above)
Comment (SamClinckspoor) [reply concourse/templates/role.yaml]: also remove the serviceaccount as it has no purpose with the role

File: concourse/templates/ingress.yaml (hunk 1 above)
Comment (SamClinckspoor) [reply concourse/templates/ingress.yaml]: I think kube-lego actually updates the ingress if it has a cert for it, but setting it like this doesn't hurt.

File: concourse/templates/web-svc.yaml (hunk 5 above)
Comment (SamClinckspoor) [reply concourse/templates/web-svc.yaml]: you don't need the tls block here.

**Comment by SamClinckspoor** [comment]
You could set the service to `type: Loadbalancer`for the ssh endpoint, that will create an ELB with TCP I think

File: concourse/templates/ingress.yaml (hunk 1 above)
Comment (iuriaranda) [reply concourse/templates/ingress.yaml]: I've tested it without the tls block and kube-lego did nothing 😕 
Maybe I did something wrong, but once I added back the tls block it worked.

File: concourse/values.yaml [hunk 6]
```diff
@@ -2,28 +2,33 @@
 ## This is a YAML-formatted file.
 ## Declare variables to be passed into your templates.
 
-## Override the name of the Chart.
+## Configuration values for Concourse.
+## ref: https://concourse.ci/setting-up.html
 ##
-# nameOverride:
+concourse:
```
Comment (SamClinckspoor) [inline concourse/values.yaml]: add `tsaPort`

**Review by SamClinckspoor — APPROVED** [review]

**Pull request merged**